src/chart_types/xy_chart/rendering/utils.ts

```diff
@@ -67,6 +67,25 @@ export function isDatumFilled({ filled, initialY1 }: DataSeriesDatum) {
 export function getClippedRanges(dataset: DataSeriesDatum[], xScale: Scale, xScaleOffset: number): ClippedRanges {
   let firstNonNullX: number | null = null;
   let hasNull = false;
+
+  const completeDatasetIsNull = dataset.filter((data) => !isDatumFilled(data)).length === 0;
+
+  if (completeDatasetIsNull)
+    return dataset.reduce<ClippedRanges>((acc, data) => {
+      const xScaled = xScale.scale(data.x);
+      if (xScaled === null) {
+        return acc;
+      }
+
+      const xValue = xScaled - xScaleOffset + xScale.bandwidth / 2;
+      const endXValue = xScale.range[1] - xScale.bandwidth * (2 / 3);
+      if (xValue === endXValue) {
+        acc.push([0, xValue]);
+      }
+
+      return acc;
+    }, []);
```

Comment:
This would work just fine, however since we know we need to clip the whole `range` we can just pull that off the `xScale`.

```suggestion
  if (completeDatasetIsNull) {
    const [min, max] = xScale.range;
    return [[min, max - xScale.step / 2]];
  }
```

I don't think we _really_ need to account for the `step` at the end since this would only apply to this series, but rather be safe than sorry.